### ISSUE TITLE
Issue with decompressFromBase64

### ISSUE THREAD

**Issue by alocher** [nullpunkt/lz-string-php#6]
I encountered a bug decompressing a string with lz-string-php that has been compressed with lz-string for JavaScript. 

The following example shows a compressed string I can successfully compress/decompress with JS but PHP throws an error. It seems to have something to do with special characters but only in certain combinations. I'm using compressToBase64 and decompressFromBase64. Following scenario shows that the compressed string is different between JS and PHP version which results in the error at the bottom if I try to uncompress a JS compressed string with PHP.

```
Uncompressed String: aä´

PHP: compressed String: IYACAA==

JS: compressed String: IYE4LUA=

PHP: uncompress compressed PHP String: a  => this shows that only a has been compressed

PHP: uncompress compressed JS String:
Fatal error: Uncaught exception 'Exception' with message 'Too much errors.' in 
/home/xyz/LZString.php:280 Stack trace: #0 /home/xyz/LZString.php(376): 
LZString::decompress('\xE2\x86\x81\xE3\xA0\xAD') #1 /home/xyz/test.php(22): 
LZString::decompressFromBase64('IYE4LUA=') #2 {main} thrown in /home/xyz/LZString.php on line 280
```

PHP Test script:
```
<?php

require("LZString.php");

ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);

$rawstr= 'aä´';

$compressedjs = 'IYE4LUA=';

$compressed = LZString::compressToBase64($rawstr);

echo 'Uncompressed String: ' . $rawstr;
echo "<br><br>";
echo 'PHP compressed String: ' . $compressed;
echo "<br><br>";
echo 'JS compressed String: IYE4LUA=';
echo "<br><br>";
echo 'PHP uncompressed String: ' .  LZString::decompressFromBase64($compressed);
echo "<br><br>";
echo 'JS uncompressed String: ' .  LZString::decompressFromBase64($compressedjs);

?>
```

**Comment by Korcholis** [comment]
To those with problems with decompressing from Base64, use [this fork](https://github.com/JuhaszAdam/lz-string-php). 

It works really fast and decompressed me a json string (done in JS and passed through Ajax) with accents and stuff, just like I asked. The only visible changes are the extra two files and the `\LZCompressor` namespace, but it comes with composer, so it's darn easy to use.

**Comment by nullpunkt** [comment]
Hi - as Korcholis said, he refactored the implementation and cleared up the code, so i merged in his fork into the master.

**Comment by Korcholis** [comment]
Oh, good then!

**Comment by nullpunkt** [comment]
Hi, i've completely rewritten the PHP component to match the current algorythm / ouput provided by the javascript component. The bug is fixed in version 1.1.0.
